x/paloma/keeper/keeper.go

```diff
@@ -161,3 +175,201 @@ func (k Keeper) CheckChainVersion(ctx context.Context) {
+}
+
+func (k Keeper) SetLightNodeClientFeegranter(
+	ctx context.Context,
+	acct sdk.AccAddress,
+) error {
+	st := k.Store(ctx)
+	key := types.LightNodeClientFeegranterKey
+
+	feegranter := &types.LightNodeClientFeegranter{
+		Account: acct,
+	}
+
+	return keeperutil.Save(st, k.cdc, key, feegranter)
+}
+
+func (k Keeper) AllLightNodeClientLicenses(
+	ctx context.Context,
+) ([]*types.LightNodeClientLicense, error) {
+	st := k.lightNodeClientLicenseStore(ctx)
+	_, all, err := keeperutil.IterAll[*types.LightNodeClientLicense](st, k.cdc)
+	return all, err
+}
+
+func (k Keeper) SetLightNodeClientLicense(
+	ctx context.Context,
+	addr string,
+	license *types.LightNodeClientLicense,
+) error {
+	st := k.lightNodeClientLicenseStore(ctx)
+	return keeperutil.Save(st, k.cdc, []byte(addr), license)
+}
+
+func (k Keeper) GetLightNodeClientLicense(
+	ctx context.Context,
+	addr string,
+) (*types.LightNodeClientLicense, error) {
+	st := k.lightNodeClientLicenseStore(ctx)
+	return keeperutil.Load[*types.LightNodeClientLicense](st, k.cdc, []byte(addr))
+}
+
+func (k Keeper) CreateLightNodeClientLicense(
+	ctx context.Context,
+	creatorAddr, clientAddr string,
+	amount sdk.Coin,
+	vestingMonths uint32,
+	feegrant bool,
+) error {
+	creatorAcct, err := sdk.AccAddressFromBech32(creatorAddr)
+	if err != nil {
+		return err
+	}
+
+	if sdk.VerifyAddressFormat(creatorAcct) != nil || !amount.IsValid() {
+		return types.ErrInvalidParameters
+	}
+
+	// Check if license exists already
+	_, err = k.GetLightNodeClientLicense(ctx, clientAddr)
+	if err == nil {
+		return types.ErrLicenseExists
+	} else if !errors.Is(err, keeperutil.ErrNotFound) {
+		// Any errors other than not found and we bail
+		return err
+	}
+
+	// The client account name, to create a new base account and/or to set up
+	// the feegrant
+	acct, err := k.accountKeeper.AddressCodec().StringToBytes(clientAddr)
+	if err != nil {
+		return err
+	}
+
+	// If the account already exists, we can't move on
+	if k.accountKeeper.HasAccount(ctx, acct) {
+		return types.ErrAccountExists
+	}
+
+	license := &types.LightNodeClientLicense{
+		ClientAddress: clientAddr,
+		Amount:        amount,
+		VestingMonths: vestingMonths,
+		Feegrant:      feegrant,
+	}
+
+	// We're here for the first time, so we create the base account now, as
+	// it will be used to register the light node client and start vesting
+	baseAccount := authtypes.NewBaseAccountWithAddress(acct)
+	baseAccount = k.accountKeeper.NewAccount(ctx, baseAccount).(*authtypes.BaseAccount)
+	k.accountKeeper.SetAccount(ctx, baseAccount)
+
+	// If license has the feegrant flag, set the allowance
+	if license.Feegrant {
+		feegranter, err := k.LightNodeClientFeegranter(ctx)
+		if err != nil {
+			if errors.Is(err, keeperutil.ErrNotFound) {
+				return types.ErrNoFeegranter
+			}
+
+			return err
+		}
+
+		// TODO - we might want to set spend and expiration limits
+		allowance := &feegrantmodule.BasicAllowance{
+			SpendLimit: nil, // Unlimited spend
```

Comment:
I lack oversight of the specs for this feature, so maybe I'm mistaken here. But shouldn't we avoid a situation where we give infinite fee grant to a random client address? 

What is this grant used for, and why don't we limit it?

Reply:
The client account will be created at the time of registration, so it needs a fee grant or it won't have enough fees to even register. Then it starts vesting, but since it's over a two year period, it may not have enough funds to delegate for awhile.

The requirement is to not worry about abuse for now, but I think we could still set some reasonable limits (on both time and amount).

We could even limit the fee grant to only the registration message, so they could register but couldn't abuse it. They would have to wait a bit to delegate, though.

@taariq Should we set some limits here?

Reply:
It doesn't have to be a super low limit either. But it shouldn't be unlimited.

Reply:
@maharifu why don't we put a limit of 1 GRAIN for fees. That should be enough to get the accounts to start. Right?

Reply:
@taariq Yes, I think that should be more than enough. I'll set it.